docs/app/____-represent-flows-as-pushdown-automata.md

```diff
@@ -0,0 +1,71 @@
+# Defining an Abstraction for Application Flows as Pushdown Automata
+
+## Context
+
+We've previously been using finite state machines to represent application flows. This has provided a number of benefits:
+
+1. Provides a declarative way to define page transitions, which reduces bugs by constraining what is possible to a less expressive language
+2. Provides a core data structure that can be augmented with additional functionality, such as visualization
+
+Implementation-wise, we've been using the [xstate library](https://xstate.js.org/).
```

Comment:
Have you came across any good libraries (any language) that could provide a good example of what implementing this might look like? I found [this one](https://github.com/caleb531/automata/blob/main/docs/pda/class-dpda.md#class-dpdapda) but its docs are still a little abstract.

Reply:
Nope I haven't really looked into libraries. Pushdown automata are used under the hood by many parsers, since they are theoretically equivalent to context-free grammars (in the same way that regular finite automata are theoretically equivalent to regular expressions). But I don't know of any general purpose libraries for them. We might not need them though, we could probably come up with our own implementation.